### DISCUSSION TITLE
Access Point Mode for Tasmota Smart devices after the WiFi fails

### DISCUSSION THREAD

**Discussion by IntCleastwood** [arendst/Tasmota#21936] [Q&A] [answered]
Hello, i did not find an answer to my question, so i ask here. I have some of my smart plugs flashed with tasmota.
I noticed today that whenever the plug is disconnect (e.g. because the wifi router is down), the plug turns back into AP mode.

I am not an expert with this, but ...

Am i right that in this case
1. Everyone in the neighborhood can connect to the Accesspoint
2. Everyone could "hijack" the device (entering credentials to let the device connect to another wifi)
3. Everyone can read config and wifi credentials that are saved on the device? At least i can see my last entered SSID and i see the password with asterisk ... but isnt it saved somewhere on the device?

Generally: When turning the Wifi AP on, someone else could connect an "hijack" it if I am not fast enough to connect?

Or am i completely wrong?

What is the general recommendation for connecting securely to the own wifi and handle the error case?

Thanks!

**Comment by sfromis** [comment]
Yes, that is highly insecure, and it looks like `WifiConfig` is 2, where 4 is generally recommended as the base value. While it is not simple to access wifi credentials, which obviously is saved, it is not impossible either, for someone diving deeper than just using the regular web UI and commands.

I'd recommend changing to `WifiConfig 4` where there is no fallback to AP mode, and in case of a need to recover the device without having Wifi available, there are some recovery features.
https://tasmota.github.io/docs/Device-Recovery/

**Comment by IntCleastwood** [comment]
Thanks for your answer, the config was indeed 2, but i never changed it, so maybe in an earlier build it was 2(?) ... 

However, since i am not aware how often this happened in the past (i think it mostly happens unnoticed or when i was not at home) i try to estimate at least the potential threat for an attacker getting into my wifi ...

So, do you know how "easy" is it to get the credentials via the open AP? I tried myself by connecting to the AP and using webcommands ... i could figure out at least the SSID ... when entering the "Password" command, it just gives 4 asterisks back ... but like you said, somewhere it has to be stored as clear text, right?

**Reply by sfromis** [reply]
Yeah, as I said, not simple by just going to the AP, as the web UI does not have a feature to show the Wifi passphrase. Getting the plain text means downloading the config and decoding the data. Not sure when mode 2 can be set, but it does include things like doing recovery to Wifi Manager.

**Reply by IntCleastwood** [reply]
ok, i tried myself with the config debug tool ... it was totally easy to read the config ... pw is clear text unfortunately ... thats really awkward, since i wasn't aware of this ... i can now just hope, nobody did notice this gap and infiltrated my wifi in the past ... tasmota version was 8.1.0.2

**Reply by sfromis** [reply]
Docs does have a warning against `WifiConfig 2`
https://tasmota.github.io/docs/Securing-your-IoT-from-hacking/#disable-unsecured-fallback-wifi-wifimanager

**Reply by IntCleastwood** [reply]
Yes, like i said, i wasn't aware of this, since i noticed this for the first time, but i know that my wifi was down very often in the last years ...

**Reply by stefanbode** [reply]
Thanks for bringing this up. This is again a tradeoff between security and comfort. You can get the same "boot in AP mode" if you just JAM the wifi frequencies. easy to do with an ESP8266. I believe as a first fix we will block the download of the config in AP mode. This is an easy fix and does not harm. But we still have the problem an attacker can use the second SSID to connect to his phone and THEN download the config file. My only idea is to wipe always the OTHER password if a SSID/password gets added/changed in AP mode

**Reply by Jason2866** [reply]
@stefanbode The fix is already there use `WifiConfig 4`. 😉
Trying to secure the AP mode gives imho only the wrong feeling "being save". There is for sure an open door left which can be used to hack. Only adding code without useable functions needing flash space.

**Reply by stefanbode** [reply]
A software should be secure by default because we also have non-neerds using Tasmota. As far as I know is wificonfig 2 the default. Maybe it is a great idea to change it to 4 after configuration. As said. I do wificonfig 4 since beginning but I am also not the standard user.

**Reply by arendst** [reply, marked as answer]
I suggest to stop this discussion. WifiConfig is 4 since many years as can be easily found in ``my_user_config.h``.

**Reply by sfromis** [reply]
No, `WifiConfig 4` is [the default](https://tasmota.github.io/docs/Commands/#wificonfig), and "reasonably safe".
